pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -570,6 +618,29 @@ func mergeMetadata(iqrs []*IQR) (*IQR, error) {
 	return result, nil
 }
 
+// Caller should validate iqr before.
+func (iqr *IQR) getRRCIQRFromWithoutRRCIQR() (*IQR, error) {
+	if iqr.mode != withoutRRCs {
+		return nil, fmt.Errorf("IQR.mergeWithoutRRCIntoRRCIQR: other mode is not withoutRRCs")
+	}
+	valuesLen := 0
+	for _, values := range iqr.knownValues {
+		valuesLen = len(values)
+		break
+	}
+	newIQR := NewIQR(iqr.qid)
+	err := newIQR.AppendRRCs(make([]*utils.RecordResultContainer, valuesLen), nil)
+	if err != nil {
+		return nil, fmt.Errorf("IQR.mergeWithoutRRCIntoRRCIQR: error appending RRCs: %v", err)
+	}
+	err = newIQR.AppendKnownValues(iqr.knownValues)
+	if err != nil {
+		return nil, fmt.Errorf("IQR.mergeWithoutRRCIntoRRCIQR: error appending known values: %v", err)
+	}
+
+	return newIQR, nil
```

Comment:
the func names in the fmt error statements is not correct. 
also the name of this func is a bit confusing, could you simplify this a bit

Reply:
addressed in the latest commit.